### DISCUSSION TITLE
Depending On Self leads to infinite loop of Requests

### DISCUSSION THREAD

**Discussion by muhammadsaeedparacha** [laravel/nova-issues#5563] [Q&A] [answered]
- Laravel Version: 9.52.7  
- Nova Version: 4.24.4
- PHP Version: 8.2.4  
- Database Driver & Version:
- Operating System and Version: Mac Latest
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Description:
// Close issue if this is outside scope, and no way of avoiding it

*Problem:*
A field depending on itself
Leads to infinite loop of dependsOn()

Field:
```
Text::make('Name', 'name')->required()
            ->dependsOn(['name'], function (Text $field, NovaRequest $request, FormData $formData) {
                $field->help(Str::singular(Str::studly($formData->name)));
            })
```

Loop:
<img width="1984" alt="Screenshot 2023-05-24 at 7 11 24 AM" src="https://github.com/laravel/nova-issues/assets/2680384/69553fe3-3474-4160-bfbe-26a3a3ee2eb0">

*Usage Reasoning*
To Show User the input will be overridden, in the following manner

**Comment by crynobone** [comment]
This definitely something that should be considered out of scope using the built-in fields.

However, it is possible that you can create a custom field and override the following method:

```js 
    emitOnSyncedFieldValueChange() {
      // this.emitFieldValueChange(this.field.attribute, this.currentField.value)
    },
```

**Comment by crynobone** [comment, marked as answer]
You should be able to depends on self in 4.26.0

**Reply by potsky** [reply]
Is it still working on 4.35.4 with a Select ?
I have infinite loop too and want to change the help text when Select value changes

```php
            Select::make(__('Source'), 'source')
                ->hideFromIndex()
                ->options(ContractSourceEnum::getList())
                ->displayUsingLabels()
                ->rules('required')
                ->required()
                ->dependsOn('source', function (Select $field, NovaRequest $request, FormData $formData) {
ray(1);
                    if ($field->value === ContractSourceEnum::CUSTOMER_RENEW->value) {
                        return;
                    }

                    $field->help(
                        sprintf(
                            '<span style="color: red">%s</span>',
                            __('This team has already signed contracts. You should only select "Customer renewal" as source.')
                        )
                    );
                }),
```
